### PR TITLE
1.17/fix/persistproxyspec

### PR DESCRIPTION
**Pull request by nfuden**
# Description
In 1.17 we introduced a regression for instances using persist proxy spec.

Persist proxy spec is mostly non-useful and will be removed in 1.18 but does have some smaller edge cases still supported in 1.17.
Unlike most of our resources proxies were called at a stage where we can accept a resource into our gloo system but fail to write any statuses back out along with failing to update the proxy CR at rest in etcd.

## Code changes
Label selection cleanup

## CI changes
Updated Upgrade tests to check for this step

# Context
On upgrading from a version 1.13+ to 1.17+ with persist proxy spec set to true manual intervention was required prior to this wrk.

## Testing steps

Thanks to sam https://github.com/solo-io/gloo-gateway-shared-resources/tree/main/issues/gloo/9968

Original reproduction was hampered by overly aggressive clean up scripts scheduled by default on local so this simplest set up helps alot.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by nfuden** [comment]
https://github.com/solo-io/gloo/actions/runs/10854177366/job/30124071891?pr=10025

[test/kube2e/upgrade: spaces not tabs](https://github.com/solo-io/gloo/pull/10025/commits/84f067673dc553cb564c54b1b1396cbdcb287fa8)

https://github.com/solo-io/gloo/pull/10025/commits/84f067673dc553cb564c54b1b1396cbdcb287fa8

Shows it failing from persist

File: projects/clusteringress/pkg/translator/translator_syncer.go [hunk 1]
```diff
@@ -51,8 +51,10 @@ var (
 	//
 	// This is only required for backwards compatibility.
 	// Once users have upgraded to a version with new labels, we can delete this code and read/write the same labels.
+	// gloo-knative-translator removed in 1.17
```
Comment (sam-heilbron) [inline projects/clusteringress/pkg/translator/translator_syncer.go:54]: 👏 super helpful!

**Review by sam-heilbron — APPROVED** [review]

File: test/kube2e/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -392,13 +412,8 @@ gatewayProxies:
...
         options:
-          dlp:
-            dlpRules:
-            - actions:
-              - actionType: KEYVALUE
-                keyValueAction:
-                  keyToMask: test
-                  name: test
+          buffer:
+            maxRequestBytes: 999999
```
Comment (jenshu) [inline test/kube2e/upgrade/upgrade_test.go:416]: what was the reason for this change?

File: changelog/v1.17.8/persist-proxy-upgrade-fix.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/gloo/issues/9968
+    resolvesIssue: false
+    description: >-
+      Fixes an issue on upgrades to 1.17+ where persistproxyspec was set to true.
```
Comment (jenshu) [inline changelog/v1.17.8/persist-proxy-upgrade-fix.yaml:6]: nit: do we want to give the precise field name, i.e. `gateway.persistProxySpec` in Settings (or the helm value), so users can check if they have it set?

**Review by jenshu — DISMISSED** [review]

File: test/kube2e/upgrade/upgrade_test.go (hunk 2 above)
Comment (nfuden) [reply test/kube2e/upgrade/upgrade_test.go:416]: dlp is ee only so we need to test something that is not ee

File: changelog/v1.17.8/persist-proxy-upgrade-fix.yaml (hunk 3 above)
Comment (nfuden) [reply changelog/v1.17.8/persist-proxy-upgrade-fix.yaml:6]: Good point updated

**Review by jenshu — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**